src/charm.py

```diff
@@ -19,12 +27,30 @@ class SSSDCharm(CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
+        self.ldapclient = LdapClientRequires(self, "ldap-client")
```

Comment:
Just a small thing. Generally I prefix integration interfaces with `_`:

```suggestion
        self._ldapclient = LdapClientRequires(self, "ldap-client")
```

Encapsulation doesn't really exist in Python, but I like to prefix with the bar `_` because it signals to developers that they shouldn't be touching this value if they do not really know what they are doing.